### PR TITLE
Refactor the kyuubi session type

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Now, we support SQL, Scala and Python.

In this pr, refactor the session type: `SQL` => `INTERACTIVE`.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
Cc @pan3793 any suggestion for this？

**Comment by pan3793** [comment]
I think the idea is good, will it introduce backward compatible issue？

**Comment by turboFei** [comment]
> I think the idea is good, will it introduce backward compatible issue？

Now the SQL sessionType only used in `KyuubiSessionEvent`.

File: docs/deployment/migration-guide.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 * In Kyuubi 1.7, `kyuubi.ha.zookeeper.engine.auth.type` does not fallback to `kyuubi.ha.zookeeper.auth.type`.  
   When Kyuubi engine does Kerberos authentication with Zookeeper, user needs to explicitly set `kyuubi.ha.zookeeper.engine.auth.type` to `KERBEROS`.
 * Since Kyuubi 1.7, Kyuubi returns engine's information for `GetInfo` request instead of server. To restore the previous behavior, set `kyuubi.server.info.provider` to `SERVER`.
+* Since Kyuubi 1.7, Kyuubi session type `SQL` is refactored to `INTERACTIVE`, because Kyuubi supports not only `SQL` session, but also `SCALA` and `PYTHON` sessions.
```
Comment (turboFei) [inline docs/deployment/migration-guide.md:25]: added into migration guide. @pan3793

Comment (pan3793) [reply docs/deployment/migration-guide.md:25]: the statement clearly explained the reason, but what should user do for this change?

Comment (turboFei) [reply docs/deployment/migration-guide.md:25]: updated

**Review by pan3793 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master

**Pull request closed without merging**